src/sync.ts

```diff
@@ -602,6 +602,9 @@ export class SyncApi {
             if (shouldClear) {
                 this.storeIsInvalid = true;
                 const reason = InvalidStoreError.TOGGLED_LAZY_LOADING;
+                // TODO - InvalidStoreError, or rather all the custom error classes, should extend
+                // the MatrixError class. For a quickfix, `error` has been casted to `any` because `updateSyncState` requires
+                // it to be an instance of MatrixError class.
```

Comment:
This is a sign that we should extend the class now, rather than assume the types will work. The function wants a specific contract to be met, and we're intentionally not providing that :(

Reply:
That would require code changes in multiple other files, and this pull request is meant for only converting error.js to Typescript, which is a cosmetic change as it doesn't change the working of the original code. Imho, this should be done in a separate pull request.

Reply:
Conversions to typescript can and should affect other files: the point is to fix bugs, not check boxes :)